### DISCUSSION TITLE
Mathematics

### DISCUSSION THREAD

**Discussion by jack-dunham** [nvim-neorg/neorg#103] [Ideas]
I thought I would create a discussion for mathematics, and how it should be included (as a module I assume?)

Org handles mathematics with latex syntax; using `\(...\)` for in line math, and `\[...]\` for display math. Optionally, one can use the traditional tex style `$...$` and `$$...$$`, however there are stricter rules for when these characters actually denote math delimiters compared to the former, recommended way. I believe the latex style is easier to parse compared to the tex style which I find much more readable considering the abundance of brackets that already exist in mathematical formulae. I guess we would also need an `@equation` tag that can be used for labelled display math. 
 
Ideally it would be nice to have the source code of the equations concealed behind a pretty rendering of the equation using unicode; unicode is quite powerful when it comes to typsetting equations. Ambitious: something like [nabla.nvim](https://github.com/jbyuki/nabla.nvim), but genuine concealment + unicode could be nice. 

What are the thoughts of everyone? I feel neorg should at least have a math deliminator. What should it be?

P.S. Love the project! I have been following for a while and hope to contribute one day, but for now I am still learning lua.

**Comment by vhyrro** [comment]
It definitely should! As of right now the general consensus is that `$$<content>$$` will not be supported since that goes against the design choices of Neorg's format. Using tags like `@math` (proposed by @mrossinek) would make things consistent with the file format itself and would allow for parameters and in turn finer control. We will allow `$inline maths$` just fine though.

By the way, your sentence is a bit ambiguous:
> I believe the latex style is easier to parse compared to the tex style which I find much more readable considering the abundance of brackets that already exist in mathematical formulae

It could be interpreted as:
- I believe the ***latex*** style is easier to parse compared to the tex style ***which i find much more readable*** (you find latex more readable)
- I believe the latex style is easier to parse compared to the ***tex style which I find find much more readable*** (you find the text style more readable)

Which one do you mean? :smile:

Also regarding the pretty rendering - I think we could make it work. It's definitely not on a close TODO list but i believe we could bend the limits of TUI pretty solidly and make it look really sweet. I wonder what performance issues would come with it though... we would have to wait and see :)

**Reply by jack-dunham** [reply]
Apologies if this is a duplicate---I am still getting to grips with the whole github-cum-forum aspect of it all. 

Also I am aware of that ambiguous sentence but I couldn't be bothered to re-edit...sorry! I mean the latter. I find the tex style more readable, and as you mention, it fits with the NFF with `@math` as opposed to `$$<display-math>$$` (which I never really use in latex code anyway.)

As for the pretty rendering, I guess one solution to performance is that of nabla.nvim, where math is *replaced* with unicode with the ability to revert back to latex. In that case, I believe it would be sensible to have an option to revert to latex on write (as in the buffer can have unicode math, but the underlying file will only ever store the corresponding latex.) Math should be stored in latex format and not unicode format for obvious reasons. 

Anyway yes, this is by no means even remotely a priority and perhaps not even in the scope of what modules should come packaged with Neorg (I guess you can be the judge of that). A `@math` tag and `$<inline>$` should certainly do. 

In the meantime I am going to play around with nabla.nvim and see what cool things I can come up with. Sympy also has a pretty print function that can output equations in terminal using unicode---I am gonna have a look at that because now I am just interesting more than anything. I don't like Org's solution of just embedding equations as images to be honest...

**Reply by robbielyman** [reply]
[VimTeX](https://github.com/lervag/vimtex) has a pretty good system going using vim's conceals to make LaTeX equations prettier; might be worth looking into.
